From 0a6103eaeb7f22c009f9add87c84780b6f7f293a Mon Sep 17 00:00:00 2001
From: Muhammad Omair Javaid <omair.javaid@linaro.org>
Date: Mon, 13 May 2024 16:01:17 +0500
Subject: [PATCH] Revert "[lldb] Attempt to fix signal-in-leaf-function-aarch64
 on darwin"

This reverts commit b903badd73a2467fdd4e363231f2bf9b0704b546.

TestInterruptBacktrace was broken on AArch64/Windows as a result of this change.
see lldb-aarch64-windows buildbot here:
https://lab.llvm.org/buildbot/#/builders/219/builds/11261
---
 .../test/Shell/Unwind/Inputs/signal-in-leaf-function-aarch64.c | 2 +-
 lldb/test/Shell/Unwind/signal-in-leaf-function-aarch64.test    | 3 ---
 2 files changed, 1 insertion(+), 4 deletions(-)

diff --git a/lldb/test/Shell/Unwind/Inputs/signal-in-leaf-function-aarch64.c b/lldb/test/Shell/Unwind/Inputs/signal-in-leaf-function-aarch64.c
index fe020affcad0..9a751330623f 100644
--- a/lldb/test/Shell/Unwind/Inputs/signal-in-leaf-function-aarch64.c
+++ b/lldb/test/Shell/Unwind/Inputs/signal-in-leaf-function-aarch64.c
@@ -7,7 +7,7 @@ int __attribute__((naked)) signal_generating_add(int a, int b) {
       "ret");
 }
 
-void sigill_handler(int signo) { _exit(0); }
+void sigill_handler(int) { _exit(0); }
 
 int main() {
   signal(SIGILL, sigill_handler);
diff --git a/lldb/test/Shell/Unwind/signal-in-leaf-function-aarch64.test b/lldb/test/Shell/Unwind/signal-in-leaf-function-aarch64.test
index 09f17c174bbf..0580d0cf734a 100644
--- a/lldb/test/Shell/Unwind/signal-in-leaf-function-aarch64.test
+++ b/lldb/test/Shell/Unwind/signal-in-leaf-function-aarch64.test
@@ -4,9 +4,6 @@
 # RUN: %clang_host %S/Inputs/signal-in-leaf-function-aarch64.c -o %t
 # RUN: %lldb -s %s -o exit %t | FileCheck %s
 
-# Convert EXC_BAD_INSTRUCTION to SIGILL on darwin
-settings set platform.plugin.darwin.ignored-exceptions EXC_BAD_INSTRUCTION
-
 breakpoint set -n sigill_handler
 # CHECK: Breakpoint 1: where = {{.*}}`sigill_handler
 
-- 
2.46.0.rc1.232.g9752f9e123-goog

